Src/LinearSolvers/MLMG/AMReX_MLTensor_3D_K.H

```diff
@@ -21,25 +659,22 @@ void mltensor_fill_corners (int icorner, Box const& vbox, // vbox: the valid box
                             Array4<Real const> const& bcvalxhi,
                             Array4<Real const> const& bcvalyhi,
                             Array4<Real const> const& bcvalzhi,
-                            GpuArray<BoundCond,2*AMREX_SPACEDIM*AMREX_SPACEDIM> const& bct,
-                            GpuArray<Real,2*AMREX_SPACEDIM*AMREX_SPACEDIM> const& bcl,
+                            Array2D<BoundCond,
+                                    0,2*AMREX_SPACEDIM,
+                                    0,AMREX_SPACEDIM> const& bct,
+                            Array2D<Real,
+                                    0,2*AMREX_SPACEDIM,
+                                    0,AMREX_SPACEDIM> const& bcl,
                             int inhomog, int maxorder,
-                            GpuArray<Real,AMREX_SPACEDIM> const& dxinv, Box const& domain) noexcept
+                            GpuArray<Real,AMREX_SPACEDIM> const& dxinv,
+                            Dim3 const& dlo, Dim3 const& dhi) noexcept
 {
-    constexpr int oxlo = 0;
-    constexpr int oylo = 1;
-    constexpr int ozlo = 2;
-    constexpr int oxhi = 3;
-    constexpr int oyhi = 4;
-    constexpr int ozhi = 5;
     constexpr int xdir = 0;
     constexpr int ydir = 1;
     constexpr int zdir = 2;
     const auto blen = amrex::length(vbox);
     const auto vlo  = amrex::lbound(vbox);
     const auto vhi  = amrex::ubound(vbox);
-    const auto dlo  = amrex::lbound(domain);
-    const auto dhi  = amrex::ubound(domain);
     for (int icomp = 0; icomp < AMREX_SPACEDIM; ++icomp) {
         switch (icorner) {
         case 0: {
```

Comment:
Doesn't this run into basically the same problem you outlined in 2D for the case of a corner with 3 no-slip walls? If `vel(i-1,j,k) = -vel(i,j,k)` etc, then we get `vel(i-1,j-1,k-1) = -5* vel(i,j,k)`